nexus/src/app/external_endpoints.rs

```diff
@@ -212,16 +256,25 @@ impl ExternalEndpoints {
 
 /// Describes a single external "endpoint", by which we mean an external DNS
 /// name that's associated with a particular Silo
-#[derive(Debug, Eq, PartialEq, Serialize)]
-struct ExternalEndpoint {
-    /// the Silo associated with this endpoint
+#[derive(Debug, PartialEq, Eq, Serialize)]
+pub struct ExternalEndpoint {
+    /// the id of the Silo associated with this endpoint
+    // This is redundant with `db_silo`, but it's convenient to put it here and
+    // it shows up in the serialized form this way.
     silo_id: Uuid,
+    /// the silo associated with this endpoint
+    #[serde(skip)]
+    db_silo: Arc<nexus_db_model::Silo>,
```

Comment:
why does this need to be Arc?

Reply:
I think it doesn't strictly need to be right now.  The intent is that we will support multiple endpoints for a Silo (e.g., a DNS alias).  Most of the code here I think is written to assume that might be the case and I didn't want to clone the whole record.